slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -578,6 +583,7 @@ void SLAMServiceImpl::ProcessDataAndStartSavingMaps(double data_start_time) {
                 tmp_global_pose = map_builder.GetGlobalPose(trajectory_id,
                                                             local_poses.back());
             }
+            optimizing = false;
```

Comment:
Same here

Reply:
I can see how using a try/catch here might make sense. This way we could continue serving requests even if the optimization fails. Is this the motivation for using a try/catch here?

If so, then my concern is that if the optimization fails, the map might be corrupted, and we'd prefer to have cartographer fail and stop serving requests after all. What do you think about that?

Reply:
The motivation for the try/catch is to make sure we still set `optimizing` to the correct value. That's true that if the exception terminates the program, then it's not important. But if the calling code catches the exception, `optimizing` should be set correctly. I think the correct way to do this is wrap in a try/catch and re-throw the original exception in the catch.

Reply:
Using a mutex instead that is released when the lock goes out of scope.